src/main/java/org/logstash/beats/Message.java

```diff
@@ -30,20 +68,23 @@ public int compareTo(Message o) {
         return Integer.compare(getSequence(), o.getSequence());
     }
 
-    public Batch getBatch() {
-        return batch;
+    public void setBatch(Batch batch){
+        this.batch = batch;
     }
 
-    public void setBatch(Batch newBatch) {
-        batch = newBatch;
+    public Batch getBatch(){
+        return batch;
     }
 
     public String getIdentityStream() {
+        if (identityStream == null){
+            identityStream = extractIdentityStream();
+        }
         return identityStream;
     }
 
     private String extractIdentityStream() {
-        Map beatsData = (HashMap<String, String>) this.getData().get("beat");
+        Map beatsData = (Map)this.getData().get("beat");
```

Comment:
@robbavey why drop the type information here? Shouldn't this be `Map<String, String>`?

Reply:
Will change